#!/bin/bash
#PBS -q normal
#PBS -l select=1:ncpus=4:mem=128G
#PBS -l walltime=24:00:00
#PBS -N RemoveBadSamplesBOL
#PBS -P Personal

# Run this script on the final, filtered files generated by PLINK during filtering of the variant call files earlier.
# Manually created a .txt file named "remove_Boleracea_samples.txt" and place it in the "$OUTDIR"
# This file contains 2 columns, same sample ID in both columns

# Define directories
PLINK_BIN="/path/to/PLINK/executable/file" 
OUTDIR="/path/for/PLINK/ESativa/output_files"

# Change to the directory containing shell scripts to collect log files
cd $PBS_O_WORKDIR

# if no log folder, then make one
[ -d log ] || mkdir log

# Remove samples using PLINK
$PLINK_BIN --bfile "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005" \
    --keep-allele-order --allow-extra-chr --allow-no-sex \
    # Ensure that the .txt file containing sample IDs for removal is in the same directory as the one containing the files from 
    # PLINK filtering previously.
    --remove "$OUTDIR/remove_Boleracea_samples.txt" \
    --make-bed --out "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed"

# Deal with duplicate variant ID names
$PLINK_BIN --bfile "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed" \
    --keep-allele-order --allow-extra-chr --allow-no-sex --set-missing-var-ids @:# --make-bed \
    --out "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed_final"

# Create a PED file from the BED file
$PLINK_BIN --bfile "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed_final" \
    --keep-allele-order --allow-no-sex --allow-extra-chr --recode --tab \
    --out "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed_final"

# Create a final VCF file from the BED file
$PLINK_BIN --bfile "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed_final" \
    --keep-allele-order --allow-no-sex --allow-extra-chr --recode vcf \
    --out "$OUTDIR/Merged_BrassicaOleracea_SNPs_geno_maf005_removed_final"